shell/packages/accounts-email-token/token_server.js

```diff
@@ -81,15 +90,34 @@ Accounts.registerLoginHandler("email", function (options) {
     };
   }
 
-  if (!consumeToken(user, options.token.trim())) {
+  const tokenString = options.token.trim();
+  const maybeToken = consumeToken(user, tokenString);
+  if (!maybeToken) {
     console.error("Token not found:", options.email);
     return {
       error: new Meteor.Error(403, "Invalid authentication code"),
     };
   }
 
+  // Attempt to decrypt the resumePath, if provided.
+  let resumePath = undefined;
+  if (maybeToken.secureBox) {
+    const box = maybeToken.secureBox;
+    if (box.version === 1) {
+      const key = crypto.pbkdf2Sync(tokenString, box.salt, V1_ROUNDS, V1_KEYSIZE, V1_HASHFUNC);
```

Comment:
Have we verified that this function doesn't start out by taking a sha256 of tokenString? Since if so, then it would of course be possible to get the same results starting from the token hash as stored in the database.